### ISSUE TITLE
Cannot convert function to arrow function - it's used before - when its not

### ISSUE THREAD

**Issue by abulka** [nicoespeon/abracadabra#595]
## Describe the bug

The extension often cannot convert functions to arrow functions claiming `it's used before` (whatever that means) - when clearly nothing has been used before and there is no apparent reason for the error.

## How to reproduce

```javascript
function mary() {
}

function XX() {
    mary();
}

function sum(x) {
    return x + 1
}
```

Click on `sum` and select `convert to arrow function`.  The extension refuses to do so saying 

    I'm sorry, I can't convert this function declaration, it's used before

## Expected behaviour

The refactoring should proceed successfully and give you something like

```js
const sum = x => {
    return x + 1
};
```

## Additional information

- Version of the extension impacted: v6.13.0

🧙‍ Interestingly, 
- by commenting out the call on line 5 `mary();` the refactoring succeeds.  
- by commenting out the entire function `XX` the refactoring succeeds. 
- by commenting out the entire function `mary` the refactoring succeeds.

**Comment by nicoespeon** [comment]
Thanks for reporting @abulka, I'll have a look at it later this week 😉

**Comment by nicoespeon** [comment]
That was an interesting one. The logic that was determining if the function to convert was used earlier in the file was not working well and thoughts these `mary()` and `XX()` to be references to the `sum()` function. I added an automated test based on your report to repro.

I was able to fix it while not breaking anything else. I supposed we are good now!

Thanks for the detailed report @abulka, that was very useful 👏
